### DISCUSSION TITLE
@prisma/client did not initialize yet

### DISCUSSION THREAD

**Discussion by zmiff** [prisma/prisma#18669] [Q&A] [answered]
Can anyone help me with this issue? prisma seems to work fine on localhost, but on netlify i get this message.

I am using nuxt 3 and prisma deploying on netlify

`"message": "@prisma/client did not initialize yet. Please run \"prisma generate\" and try to import it again.\nIn case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues",`

I tried everything on the ntelify deploy guide https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-netlify.

schema.prisma:
```
datasource db {
    url      = env("DATABASE_URL")
    provider = "postgresql"
}

generator client {
    provider      = "prisma-client-js"
    binaryTargets = ["native", "rhel-openssl-1.0.x"]
}

```

package.json
```
{
  "private": true,
  "prisma": {
    "seed": "node prisma/seed.js"
  },
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview"
  },
  "devDependencies": {
    "@nuxtjs/supabase": "^0.3.1",
    "@nuxtjs/tailwindcss": "^6.1.3",
    "@tailwindcss/typography": "^0.5.8",
    "@vueuse/core": "^9.12.0",
    "@vueuse/nuxt": "^9.12.0",
    "nuxt": "3.0.0"
  },
  "dependencies": {
    "pnpm": "^7.27.0",
    "@prisma/client": "^4.7.1"
  }
}

```

API endpoint.
server/api/[books].get.ts
```
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

export default defineEventHandler(async (event) => {
  const { bookSlug } = event.context.params;

  return prisma.lesson.findFirst({
    where: {
      Books: {
        slug: bookSlug,
      },
    },
    include: {
      Book: {
        select: {
          slug: true,
          title: true,
        },
      },
    },
  });
});
```

**Comment by mohammed-bahumaish** [comment]
have you set a script to generate prisma client ? `"build": "prisma generate" `
take a look at this example https://github.com/prisma/deployment-example-netlify/blob/main/package.json

**Reply by zmiff** [reply]
Thanks for answering, well i tried to make the Build command on netlify as so "npm run build && prisma generate" nad "prisma generate" this build command fails when building 

`7:35:14 PM:   Netlify Build                                                 
7:35:14 PM: ────────────────────────────────────────────────────────────────
7:35:14 PM: ​
7:35:14 PM: ❯ Version
7:35:14 PM:   @netlify/build 29.9.0
7:35:14 PM: ​
7:35:14 PM: ❯ Flags
7:35:14 PM:   baseRelDir: true
7:35:14 PM:   buildId: 
7:35:14 PM:   deployId: 
7:35:14 PM: ​
7:35:14 PM: ❯ Current directory
7:35:14 PM:   /opt/build/repo
7:35:14 PM: ​
7:35:14 PM: ❯ Config file
7:35:14 PM:   No config file was defined: using default values.
7:35:14 PM: ​
7:35:14 PM: ❯ Context
7:35:14 PM:   production
7:35:14 PM: ​
7:35:14 PM:   1. Build command from Netlify app                             
7:35:14 PM: ────────────────────────────────────────────────────────────────
7:35:14 PM: ​
7:35:14 PM: $ prisma generate
7:35:14 PM: bash: prisma: command not found
7:35:14 PM: ​
7:35:14 PM:   "build.command" failed                                        
7:35:14 PM: ────────────────────────────────────────────────────────────────
7:35:14 PM: ​
7:35:14 PM:   Error message
7:35:14 PM:   Command failed with exit code 127: prisma generate (https://ntl.fyi/exit-code-127)
7:35:14 PM: ​
7:35:14 PM:   Error location
7:35:14 PM:   In Build command from Netlify app:
7:35:14 PM:   prisma generate`

**Reply by mohammed-bahumaish** [reply]
can you try `npm run build && npx prisma generate`. let me know if it works.

**Reply by zmiff** [reply]
This build command (npm run build && npx prisma generate) worked just fine. But i still get the same error when trying to fetch data from endpoint.

`"message": "@prisma/client did not initialize yet. Please run \"prisma generate\" and try to import it again.\nIn case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues",`

`8:02:59 PM:   Netlify Build                                                 
8:02:59 PM: ────────────────────────────────────────────────────────────────
8:02:59 PM: ​
8:02:59 PM: ❯ Version
8:02:59 PM:   @netlify/build 29.9.0
8:02:59 PM: ​
8:02:59 PM: ❯ Flags
8:02:59 PM:   baseRelDir: true
8:02:59 PM:   buildId: 6431ac3db205786089c260b0
8:02:59 PM:   deployId: 6431ac3db205786089c260b2
8:02:59 PM: ​
8:02:59 PM: ❯ Current directory
8:02:59 PM:   /opt/build/repo
8:02:59 PM: ​
8:02:59 PM: ❯ Config file
8:02:59 PM:   No config file was defined: using default values.
8:02:59 PM: ​
8:02:59 PM: ❯ Context
8:02:59 PM:   production
8:02:59 PM: ​
8:02:59 PM:   1. Build command from Netlify app                             
8:02:59 PM: ────────────────────────────────────────────────────────────────
8:02:59 PM: ​
8:02:59 PM: $ npm run build && npx prisma generate
8:02:59 PM: > build
8:02:59 PM: > nuxt build
8:02:59 PM: [log] Nuxi 3.0.0
8:02:59 PM: [log] Nuxt 3.0.0 with Nitro 1.0.0
8:03:00 PM: [info] [nuxt:tailwindcss] Using default Tailwind CSS file from runtime/tailwind.css
8:03:03 PM: [info] Client built in 2442ms
8:03:03 PM: [info] Building server...
8:03:05 PM: [success] Server built in 1654ms
8:03:05 PM: [success] [nitro] Generated public dist
8:03:05 PM: [info] [nitro] Building Nitro Server (preset: `netlify`)
8:03:07 PM: [warn] Multiple major versions of package `ufo` are being externalized. Picking latest version:
8:03:07 PM:   - /opt/build/repo/node_modules/ufo@0.8.6
8:03:07 PM:   + /opt/build/repo/node_modules/nitropack/node_modules/ufo@1.0.1
8:03:07 PM: [success] [nitro] Nitro server built
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/server.mjs (393 B) (243 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/package.json (944 B) (403 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/some.mjs.map (193 B) (150 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/some.mjs (173 B) (156 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/rollup/_virtual_head-static.mjs.map (111 B) (112 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/rollup/_virtual_head-static.mjs (301 B) (217 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/nitro/netlify.mjs.map (23 kB) (3.98 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/nitro/netlify.mjs (19.6 kB) (5.96 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/meta.get.mjs.map (3.6 kB) (542 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/meta.get.mjs (6.53 kB) (1.2 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/handlers/renderer.mjs.map (19.9 kB) (3.46 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/handlers/renderer.mjs (15.9 kB) (4.91 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/error-500.mjs.map (1.04 kB) (371 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/error-500.mjs (4.26 kB) (1.82 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/styles.mjs.map (152 B) (136 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/styles.mjs (86 B) (93 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/server.mjs.map (351 kB) (42.1 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/server.mjs (283 kB) (57.6 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/client.manifest.mjs.map (3.29 kB) (586 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/client.manifest.mjs (4.06 kB) (701 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/useSupabaseClient.b427f5eb.mjs.map (955 B) (427 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/useSupabaseClient.b427f5eb.mjs (899 B) (421 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/useFirstLesson.593b9ecb.mjs.map (343 B) (230 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/useFirstLesson.593b9ecb.mjs (257 B) (196 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/login.561877d1.mjs.map (1.35 kB) (561 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/login.561877d1.mjs (2.19 kB) (982 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/index.c6a5b8d2.mjs.map (1.41 kB) (574 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/index.c6a5b8d2.mjs (2.16 kB) (900 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/error-component.46721248.mjs.map (2.01 kB) (736 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/error-component.46721248.mjs (2.88 kB) (1.08 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/default.e4ecbe8f.mjs.map (926 B) (413 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/default.e4ecbe8f.mjs (1.56 kB) (699 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/custom.15d0d9f7.mjs.map (922 B) (409 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/custom.15d0d9f7.mjs (1.5 kB) (666 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/course.c6a04d38.mjs.map (5.84 kB) (1.55 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/course.c6a04d38.mjs (7.25 kB) (2.19 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/auth.90d29786.mjs.map (478 B) (281 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/auth.90d29786.mjs (1.09 kB) (507 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/404.2b87d9ae.mjs.map (784 B) (372 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/404.2b87d9ae.mjs (1.44 kB) (665 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/_lessonSlug_.81d18445.mjs.map (5.77 kB) (1.45 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/app/_nuxt/_lessonSlug_.81d18445.mjs (6.26 kB) (1.9 kB gzip)
8:03:08 PM: 
8:03:08 PM: [log]   ├─ .netlify/functions-internal/server/chunks/_lessonSlug_.get.mjs.map (640 B) (281 B gzip)
8:03:08 PM: 
8:03:08 PM: [log]   └─ .netlify/functions-internal/server/chunks/_lessonSlug_.get.mjs (568 B) (306 B gzip)
8:03:08 PM: 
8:03:08 PM: [log] Σ Total size: 3.67 MB (830 kB gzip)
8:03:09 PM: npm WARN exec The following package was not found and will be installed: prisma@4.12.0
8:03:10 PM: Prisma schema loaded from prisma/schema.prisma
8:03:11 PM: ✔ Generated Prisma Client (4.12.0 | library) to ./node_modules/@prisma/client in 63ms
8:03:11 PM: You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
8:03:11 PM: ```
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
```
8:03:11 PM: ​
8:03:11 PM: (build.command completed in 12.2s)
8:03:11 PM: ​
8:03:11 PM:   2. Functions bundling                                         
8:03:11 PM: ────────────────────────────────────────────────────────────────
8:03:11 PM: ​
8:03:11 PM: Packaging Functions from .netlify/functions-internal directory:
8:03:11 PM:  - server/server.js
8:03:11 PM: ​
8:03:13 PM: ​
8:03:13 PM: (Functions bundling completed in 1.6s)
8:03:13 PM: ​
8:03:13 PM:   3. Deploy site                                                
8:03:13 PM: ────────────────────────────────────────────────────────────────
8:03:13 PM: ​
8:03:13 PM: Starting to deploy site from 'dist'
8:03:13 PM: Calculating files to upload
8:03:13 PM: 0 new files to upload
8:03:13 PM: 1 new functions to upload
8:03:20 PM: Section completed: deploying
8:03:20 PM: Site deploy was successfully initiated
8:03:20 PM: Starting post processing
8:03:20 PM: ​
8:03:20 PM: (Deploy site completed in 6.6s)
8:03:20 PM: ​
8:03:20 PM: Post processing - HTML
8:03:20 PM:   Netlify Build Complete              `

**Reply by conorseed** [reply, marked as answer]
Try `npx prisma generate && npm run build` instead. Prisma should generate the client before nuxt tries building 😄

**Reply by zmiff** [reply]
Yes! this worked thank you :)

**Comment by csjophon** [comment]
Hello, has this problem been solved? I called npx prisma generate successfully, but the nginx reverse proxy still failed to be deployed on the server.
